pkg/admission/util_test.go

```diff
@@ -38,28 +39,84 @@ func createConfigWithOverrides(overrides map[string]string) *conf.AdmissionContr
 	return conf.NewAdmissionControllerConf([]*v1.ConfigMap{nil, {Data: overrides}})
 }
 
-// nolint: funlen
-func TestUpdatePodLabelForAdmissionController(t *testing.T) {
-	// verify when appId/queue are not given,
-	pod := &v1.Pod{
+func createMinimalTestingPod() *v1.Pod {
+	return &v1.Pod{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Pod",
 			APIVersion: "v1",
 		},
-		ObjectMeta: metav1.ObjectMeta{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec:       v1.PodSpec{},
+		Status:     v1.PodStatus{},
+	}
+}
+
+func createTestingPodWithMeta() *v1.Pod {
+	pod := createMinimalTestingPod()
+
+	pod.ObjectMeta =
+		metav1.ObjectMeta{
 			Name:            "a-test-pod",
 			Namespace:       "default",
 			UID:             "7f5fd6c5d5",
 			ResourceVersion: "10654",
 			Labels: map[string]string{
 				"random": "random",
 			},
-		},
-		Spec:   v1.PodSpec{},
-		Status: v1.PodStatus{},
-	}
+		}
 
-	if result := updatePodLabel(pod, "default"); result != nil {
+	return pod
+}
+
+func createTestingPodWithCustomFields(ns string) *v1.Pod {
+	pod := createMinimalTestingPod()
+	pod.ObjectMeta =
+		metav1.ObjectMeta{
+			Namespace: ns,
+			UID:       "abcd1234-5678-efgh-90ij-klmnopqrstuv",
+		}
+
+	return pod
+}
+
+func createTestingPodWithAppId() *v1.Pod {
+	pod := createTestingPodWithMeta()
+	pod.ObjectMeta.Labels["applicationId"] = "app-0001"
+
+	return pod
+}
+
+func createTestingPodWithGenerateName() *v1.Pod {
+	pod := createMinimalTestingPod()
+	pod.ObjectMeta.GenerateName = "some-pod-"
+
+	return pod
+}
+
+func createTestingPodWithQueue() *v1.Pod {
+	pod := createTestingPodWithMeta()
+	pod.ObjectMeta.Labels["queue"] = "root.abc"
+
+	return pod
+}
+
+func createTestingPodNoNamespaceAndLabels() *v1.Pod {
+	pod := createMinimalTestingPod()
+	pod.ObjectMeta =
+		metav1.ObjectMeta{
+			Name:            "a-test-pod",
+			UID:             "7f5fd6c5d5",
+			ResourceVersion: "10654",
+		}
+	return pod
+}
+
+// nolint: funlen
```

Comment:
we should not need this anymore function is now only 65 lines

Reply:
good catch, updated. However, found 1 warning in ``admission_controller_test``. I've added it there.